### PR TITLE
ci: exclude dev release notes from documentation labeling.

### PR DESCRIPTION
**Pull request by ioga**
## Description

it seems docs label / PR request is applied to release notes which seems excessive and unnecessary, since the release notes get through its own round of review on every release. I suspect it should be more efficient to review them in bulk once every two weeks.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by rb-determined-ai** [comment]
FWIW, our former tech writer was of the opinion that it was actually much easier to review them one at a time.  Not sure how @tara-det-ai feels about it though.

**Comment by dannysauer** [comment]
The overall idea with fully populating codeowners is that _somebody_ should be responsible for every part of the code.  In this specific case, infra has already worked with @determined-ai/docs to somewhat tune how they're alerted, so removing both the label and notification group limits their option to be aware of PRs which affect documentation.  I could see switching the release notes code owner entry to the docs-notify group, making it more clear that review of the PR is more optional, but I kinda dislike the idea of also removing the label.  I think we should be more heavily using labels, personally.  They can be super handy in searches. :)

I'll leave it up to @tara-det-ai to make the final decision, though.

**Review by tara-det-ai — COMMENTED** [review]
- ok to switch the release notes code owner entry to the docs-notify group, making it more clear that review of the PR is more optional,
- keep the label rather than remove it

File: .github/CODEOWNERS
Comment (dannysauer) [inline .github/CODEOWNERS:9]:
```suggestion
/docs/release-notes/    @determined-ai/docs-notify
```

File: .github/labeler.yml
Comment (dannysauer) [inline .github/labeler.yml:5]:
```suggestion
  - docs/**
```

**Review by dannysauer — CHANGES_REQUESTED** [review]
Proposed changes to reflect docs team's request.

**Pull request closed without merging**
